### PR TITLE
Add key binding for cljr-extract-def and prefix declarations for clojure

### PR DESCRIPTION
**Pull request by mbertheau**
This adds a key binding for one function that was missing from the list. It also sorts them alphabetically.

For some reason the prefix declarations don't show up for me.

### CODE REVIEW

**Comment by mbertheau** [comment]
I updated all packages and now they show up. <kbd>,</kbd> is broken in clojure, but that's probably unrelated.

**Comment by mbertheau** [comment]
<kbd>,</kbd> is broken only in the first clojure buffer. An all clojure buffers created after that it works. However, the prefixes from this PR show up only if I use <kbd>SPC m r</kbd> to go there; with <kbd>, r</kbd> all key descriptions are just 'prefix'.

**Comment by syl20bnr** [comment]
I merged #2889, can you rebase your changes on top of it (the merged PR already named some prefixes). Thank you !

**Comment by mbertheau** [comment]
@syl20bnr Done :sandal:

**Comment by mbertheau** [comment]
Rebased to current develop.

File: layers/+lang/clojure/packages.el [hunk 1]
```diff
@@ -316,11 +316,12 @@ If called with a prefix argument, uses the other-window instead."
           "mrcn" 'cljr-clean-ns
           "mrcp" 'cljr-cycle-privacy
           "mrdk" 'cljr-destructure-keys
-          "mref" 'cljr-extract-function
           "mrec" 'cljr-extract-constant
+          "mref" 'cljr-extract-def
+          "mref" 'cljr-extract-function
```
Comment (malchmih) [inline layers/+lang/clojure/packages.el]: Both have the same binding?

Comment (mbertheau) [inline layers/+lang/clojure/packages.el]: Oops, should be `mred`, thanks.

I'm on vacation now and can't change the PR so feel free to fix that during
merging.

Malchevskiy Misha <notifications@github.com> schrieb am Mi., 30. Sep. 2015
14:02:

**Comment by TheBB** [comment]
Thanks! Cherry-picked in develop, you can safely delete your branch. I added a commit to fix the binding and add documentation: https://github.com/syl20bnr/spacemacs/commit/e8f146acfbc1e8201f78141df2aec23ca307a685 and another one to fix my documentation again ;-) https://github.com/syl20bnr/spacemacs/commit/f4b3f439616838527367dda5977b991273731fee

**Pull request closed without merging**